platform/utm_mgrs_utils.cpp

```diff
@@ -0,0 +1,567 @@
+#include "platform/utm_mgrs_utils.hpp"
+
+#include "base/string_utils.hpp"
+
+#include <string>
+#include <iostream>
+#include <cmath>
+
+namespace utm_mgrs_utils
+{
+using namespace std;
+
+typedef struct UTMPoint_Value
+{
+  double easting;
+  double northing;
+  int zone_number;
+  char zone_letter;
+
+} UTMPoint;
+
+UTMPoint latlon_to_utm(double lat, double lon);
+int latlon_to_zone_number(double lat, double lon);
+char latitude_to_zone_letter(double lat);
+int zone_number_to_central_longitude(int zone_number);
+double mod_angle(double value);
+
+string utm_to_str(UTMPoint point);
+string utm_to_mgrs_str(UTMPoint point, int precision);
+int zone_to_100k(int i);
+string get_100k_id(double easting, double northing, int zone_number);
+double square_char_to_easting(char e, int set);
+double square_char_to_northing(char n, int set);
+double zone_to_min_northing(char zoneLetter);
+
+const double PI = 3.14159165358979;
```

Comment:
Please use constexpr everywhere where possible.

Reply:
math::pi ?

Reply:
Added `constexpr` all over the code. Replaced `PI` with `math::pi`.